test_erezutils.py

```diff
@@ -0,0 +1,97 @@
+import hashlib
+import tempfile
+import unittest
+
+from erezutils import chunks, hashfiles, rupdate
+
+
+class RecursiveUpdateTest(unittest.TestCase):
+    def test_update_empty(self):
+        self.assertEqual({"a": 1}, rupdate({}, {"a": 1}))
+
+    def test_update_existing(self):
+        self.assertEqual({"a": 2}, rupdate({"a": 1}, {"a": 2}))
+
+    def test_update_with_other_keys(self):
+        self.assertEqual({"a": 2, "b": 1}, rupdate({"a": 1}, {"a": 2, "b": 1}))
+
+    def test_update_leaves_other_keys_untouched(self):
+        self.assertEqual({"a": 2, "b": 1}, rupdate({"a": 1, "b": 1}, {"a": 2}))
+
+    def test_recursive_update(self):
+        self.assertEqual({
+            "a": {"b": 2}},
+            rupdate({"a": {"b": 1}}, {"a": {"b": 2}})
+        )
+
+    def test_recursive_update_on_empty_dict(self):
+        self.assertEqual({"a": {"b": 2}}, rupdate({}, {"a": {"b": 2}}))
+
+    def test_recursive_update_with_other_keys(self):
+        self.assertEqual(
+            {"a": {"b": 2, "c": 1}},
+            rupdate({"a": {"b": 1}}, {"a": {"b": 2, "c": 1}})
+        )
+
+    def test_recursive_update_leaves_other_keys_untouched(self):
+        self.assertEqual(
+            {"a": {"b": 2, "c": 1}},
+            rupdate({"a": {"b": 1, "c": 1}}, {"a": {"b": 2}})
+        )
+
+
+class HashfilesTest(unittest.TestCase):
+    def test_hash_default_sha512(self):
+        data = b"thisisatest"
+        data_sha512 = hashlib.sha512()
+        data_sha512.update(data)
+
+        with tempfile.TemporaryFile(mode="w+b") as f:
+            f.write(data)
+            f.seek(0)
+            self.assertEqual(data_sha512.hexdigest(), hashfiles([f]))
+
+    def test_hash_sha256(self):
+        data = b"thisisatest"
+        data_sha256 = hashlib.sha256()
+        data_sha256.update(data)
+
+        with tempfile.TemporaryFile(mode="w+b") as f:
+            f.write(data)
+            f.seek(0)
+            self.assertEqual(
+                data_sha256.hexdigest(), hashfiles([f], algorithm="sha256")
+            )
+
+    def test_hash_multiple_files(self):
+        data1 = b"thisisatest"
+        data2 = b"andanothertest"
+        data_sha512 = hashlib.sha512()
+        data_sha512.update(data1)
+        data_sha512.update(data2)
+
+        with tempfile.TemporaryFile(mode="w+b") as f1:
+            f1.write(data1)
+            f1.seek(0)
+            with tempfile.TemporaryFile(mode="w+b") as f2:
+                f2.write(data2)
+                f2.seek(0)
+                self.assertEqual(data_sha512.hexdigest(), hashfiles([f1, f2]))
+
+
+class ChunksTest(unittest.TestCase):
+    def test_empty(self):
+        data = []
+        self.assertEqual([], list(chunks(data, 1)))
+
+    def test_one_chunk(self):
+        data = [1]
+        self.assertEqual([[1]], list(chunks(data, 1)))
```

Comment:
Maybe use `assertSequenceEqual` rather than post processing the result:

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSequenceEqual

Reply:
`assertSequenceEqual` does not evaluate generators.


```                                                                                           
    self.assertSequenceEqual([], chunks(data, 1))
AssertionError: Second sequence has no length.    Non-sequence?
- []
+ <generator object chunks.<locals>.<genexpr> at 0x68b777772b10> 
```

Reply:
Ah interesting. I see.